RB: add query detecting validators that use badly anchored regular expressions on library/remote input (#11824)
github/codeql: ruby/ql/src/change-notes/2023-01-06-badly-anchored-regex.md

---
category: newQuery
---

> * Added a new query, `rb/regex/badly-anchored-regexp` to detect regular expression validators that use `^` and `$` 

Reviewer: ```suggestion
* Added a new query, `rb/regex/badly-anchored-regexp`, to detect regular expression validators that use `^` and `$` 
```

Typo fix

>   as anchors and thus might match a single line instead of the entire string.

Reviewer: ```suggestion
  as anchors and therefore might match only a single line of a multi-line string.
```

Small styling tweak
